File: frontend/lib/models/member.dart
import 'package:mess/models/user.dart';

/// A member is a Person that member of a group
/// That Person may or may not have a real user account associated to them
final class Member {
  final String uuid;
  final String name;
  final User? associatedUser;

  Member(this.uuid, this.name, {this.associatedUser});
}

File: backend/Data/Member.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Mess.Data;

public class Member
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Uuid { get; set; }
    public string Name { get; set; } = "Unknown";
    public User? AssociatedUser { get; set; }
}